### PR TITLE
Add osrf_pycommon and catkin_tools [ROS]

### PR DESCRIPTION
**Pull request by wolfv**
I tried to make catkin_tools a noarch pacakge, however, it didn't work. The following multi-level nested `entry_points` seemed to have tripped up the conda-build command.
Does someone have experience with that? I tried to find other recipes but most seem to only use the implicit console_scripts?...

Cheers!

```
  noarch: python
  entry_points:
    console_scripts:
        - catkin = catkin_tools.commands.catkin:main
    catkin_tools.commands.catkin.verbs:
        - build = catkin_tools.verbs.catkin_build:description
        - clean = catkin_tools.verbs.catkin_clean:description
        - config = catkin_tools.verbs.catkin_config:description
        - create = catkin_tools.verbs.catkin_create:description
        - env = catkin_tools.verbs.catkin_env:description
        - init = catkin_tools.verbs.catkin_init:description
        - list = catkin_tools.verbs.catkin_list:description
        - locate = catkin_tools.verbs.catkin_locate:description
        - profile = catkin_tools.verbs.catkin_profile:description
    catkin_tools.jobs:
        - catkin = catkin_tools.jobs.catkin:description
        - cmake = catkin_tools.jobs.cmake:description
    catkin_tools.spaces:
        - build = catkin_tools.spaces.build:description
        - devel = catkin_tools.spaces.devel:description
        - install = catkin_tools.spaces.install:description
        - log = catkin_tools.spaces.log:description
        - source = catkin_tools.spaces.source:description
```

### CODE REVIEW

File: recipes/osrf_pycommon/recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
+
+requirements:
+  host:
+    - python
+    - pip
+    - setuptools
+  run:
+    - python
+    - setuptools
```
Comment (marcelotrevisani) [inline recipes/osrf_pycommon/recipe/meta.yaml]: Do you really need ``setuptools`` to run this project?

File: recipes/catkin_tools/recipe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+requirements:
+  host:
+    - python
+    - pip
+    - setuptools
+  run:
+    - python
+    - catkin_pkg >0.2.9
+    - setuptools
```
Comment (marcelotrevisani) [inline recipes/catkin_tools/recipe/meta.yaml]: Same here

File: recipes/osrf_pycommon/recipe/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    - trollius  # [py < 34]
+
+test:
+  imports:
+    - osrf_pycommon
+
+about:
+  home: https://github.com/osrf/osrf_pycommon
+  license: Apache-2.0
```
Comment (marcelotrevisani) [inline recipes/osrf_pycommon/recipe/meta.yaml]:
```suggestion
  license: Apache-2.0
  license_family: Apache
```

File: recipes/catkin_tools/recipe/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+test:
+  commands:
+    - catkin --help
+    - catkin init --help
+    - catkin build --help
+    - catkin list --help
+    - catkin env --help
+about:
+  home: https://catkin-tools.readthedocs.io/
+  license: Apache-2.0
```
Comment (marcelotrevisani) [inline recipes/catkin_tools/recipe/meta.yaml]:
```suggestion
  license: Apache-2.0
  license_family: Apache
```

**Comment by wolfv** [comment]
fixed! thanks for the review.

**Pull request merged**